src/main/java/com/google/sps/data/DatastoreManagerInterface.java

```diff
@@ -0,0 +1,28 @@
+package com.google.sps.data;
+
+import java.util.Date;
+import com.google.appengine.api.datastore.Entity;
+import java.util.Optional;
+import java.util.List;
+
+/** Interface for the DatastoreManager class. */
+public interface DatastoreManagerInterface {
+    void addNewAttendee(String screenName, String sessionId);
+    void addNewInstance(String instanceName, Optional<String> sessionId);
+    void addNewSession(String sessionId,
+        Optional<String> screenNameOfController, Optional<String> ipOfVM);
+    void updateAttendee(String screenName, String sessionId);
+    void updateStateOfInstance(String instanceName, String stateOfInstance);
+    void updateSessionIdOfInstance(String instanceName, 
+        Optional<String> sessionId);
+    void updateControllerOfSession(String sessionId,
+        String screenNameOfController);
+    void updateIpOfVM(String sessionId, Optional<String> ipOfVM);
+    void endSession(String sessionId);
+    void stopInstance(String instanceName);
+    void removeAttendee(String screenName, String sessionId);
+    List<String> getListOfAttendeesInSession(String sessionId);
+    List<String> getListOfInstance();
```

Comment:
What is this supposed to do? Do you mean getListOfInstances?

Reply:
It should return a list of instances and their state to see if there is an available instance to reuse or if one should be created.

Reply:
Rename it to getAvailableInstances() and have it simply return instances that dont have sessions attached to them and are TERMINATED, based on https://cloud.google.com/compute/docs/instances/instance-life-cycle. 
This assumes we shutdown instances after a session is done.